File: Aigilas/Aigilas/Entities/EntityFactory.cs
using System;
using SPX.Entities;
using SPX.Core;

namespace Agilas.Entities
{
    public class EntityFactory
    {
        public static Entity Create(int type, Point2 location)
        {
            switch (type)
            {
                case Agilas.EntityType.FLOOR:
                    return new Floor(location);
                case Agilas.EntityType.WALL:
                    return new Wall(location);
                case Agilas.EntityType.DOWNSTAIRS:
                    return new Downstairs(location);
                case Agilas.EntityType.UPSTAIRS:
                    return new Upstairs(location);
                default:
                    throw new Exception("An undefined int case was passed into the EntityFactory.");
            }
        }
    }
}

File: aigilas/src/aigilas/entities/EntityFactory.java
package aigilas.entities;

import aigilas.Common;
import sps.bridge.EntityType;
import sps.bridge.EntityTypes;
import sps.core.Core;
import sps.core.Point2;
import sps.entities.Entity;

public class EntityFactory {
    public static Entity create(EntityType type, Point2 location) {
        if (EntityTypes.get(Core.Floor) == type) {
            return new Floor(location);
        }
        if (EntityTypes.get(Common.Entities.Wall) == type) {
            return new Wall(location);
        }
        if (EntityTypes.get(Common.Entities.Downstairs) == type) {
            return new Downstairs(location);
        }
        if (EntityTypes.get(Common.Entities.Upstairs) == type) {
            return new Upstairs(location);
        }
        return null;
    }
}
